### ISSUE TITLE
Modules & imports

### ISSUE THREAD

**Issue by rolyp** [explorable-viz/fluid#571]
Some kind of simple import mechanism, now that we have a non-trivial number of built-in functions. In particular the dictionary operations (#570) overlap with some names in the “prelude”.

**Comment by jacobpake** [comment]
As discussed we will tackle in two parts:

1. replace `FigSpec` imports with source file imports:

- [x] Add ability to import modules from source file (only programs with expressions to start)
- [x] Enforce module file names valid identifiers (i.e. `non-renewables.fld` becomes `nonRenewables.fld`
- [x] Replace `FigSpec` imports with source imports across tests, websites
- [x] Remove `imports` from `FigSpec`

2. allow imports from modules with proper environment handling and cycle detection.

---

It would also be nice to have: 

- specified function imports
- replace `fluid/lib` with `fluid-lib` as a default `fluidSrcPath` to avoid repeated `import lib.x`

**Comment by rolyp** [comment]
Once this is done I think we can bump the minor revision, i.e. go to 0.8.0.

**Comment by rolyp** [comment]
Will go for the revision bump later this week (perhaps when @Ali6461's work is merged as that seems like a nice complementary feature).
